### PR TITLE
SanityCheckerV2

### PR DESCRIPTION
**Pull request by mhn65536**

### CODE REVIEW

File: FF1Lib/Flags.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ namespace FF1Lib
 {
 	public class Flags : IIncentiveFlags, IMapEditFlags, IScaleFlags, IFloorShuffleFlags
 	{
-		public bool Spoilers { get; set; } = false;
```
Comment (latzio) [inline FF1Lib/Flags.cs:16]: This should probably stay false by default, but I realize you probably did it for testing.

File: FF1Lib/IncentiveData.cs [hunk 2]
```diff
@@ -7,8 +7,12 @@ namespace FF1Lib
 {
 	public class IncentiveData
 	{
-		public IncentiveData(MT19337 rng, IIncentiveFlags flags, OverworldMap map, ItemShopSlot shopSlot)
+		ISanityChecker _checker;
+
+		public IncentiveData(MT19337 rng, IIncentiveFlags flags, OverworldMap map, ItemShopSlot shopSlot, ISanityChecker checker)
```
Comment (latzio) [inline FF1Lib/IncentiveData.cs:12]: Nice interface injection.

File: FF1Lib/Sanity/NoYouDonTException.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace FF1Lib.Sanity
+{
+	[Serializable]
+	internal class NoYouDonTException : Exception
```
Comment (latzio) [inline FF1Lib/Sanity/NoYouDonTException.cs:7]: Don't with a capital `T` really?

File: FF1Randomizer/MainWindow.xaml.cs [hunk 4]
```diff
@@ -2,6 +2,7 @@
 using Microsoft.Win32;
 using RomUtilities;
 using System;
+using System.Diagnostics;
```
Comment (latzio) [inline FF1Randomizer/MainWindow.xaml.cs:5]: These changes are mostly meaningless.

**Review by latzio — COMMENTED** [review]
This is a heck of an effort!

File: FF1Randomizer/MainWindow.xaml.cs (hunk 4 above)
Comment (mhn65536) [reply FF1Randomizer/MainWindow.xaml.cs:5]: was required for Stopwatch, i will remove theese, when a merge draws nearer.

File: FF1Lib/Sanity/NoYouDonTException.cs (hunk 3 above)
Comment (mhn65536) [reply FF1Lib/Sanity/NoYouDonTException.cs:7]: I added this one, when i first realized there were 3500 teleporters in coneria. I'm sure we find a solution for the T.

File: FF1Lib/Flags.cs (hunk 1 above)
Comment (mhn65536) [reply FF1Lib/Flags.cs:16]: I'll go through all changed files, when a merge draws closer and revert my test thingies.

**Review by latzio — APPROVED** [review]

**Pull request merged**